pkg/agent/proxy/proxier.go

```diff
@@ -933,6 +928,9 @@ func (p *proxier) updateServiceExternalAddresses(pSvcInfo, svcInfo *types.Servic
 	if p.proxyLoadBalancerIPs {
 		deletedLoadBalancerIPs := smallSliceDifference(pSvcInfo.LoadBalancerVIPs(), svcInfo.LoadBalancerVIPs())
 		addedLoadBalancerIPs := smallSliceDifference(svcInfo.LoadBalancerVIPs(), pSvcInfo.LoadBalancerVIPs())
+
+		p.deleteServiceIPs(types.GenerateServiceInfoStrings(pSvcInfo.Protocol(), pSvcInfo.Port(), deletedLoadBalancerIPs))
```

Comment:
what about the new IPs (`addedLoadBalancerIPs`)? Are they added in a different way?

also, do we need similar handling for external IPs?

Reply:
We should also handle the external IPs updating if we add the key-value of `externalIP->svcInfo` to `ipToServiceMap`